datafusion/common/src/dfschema.rs

```diff
@@ -1323,6 +1323,49 @@ impl SchemaExt for Schema {
     }
 }
 
+/// Helper to hold table schema information.
+///
+/// A table schema consists of:
+/// - file schema: the schema of the data files
+/// - table partition columns: the columns used for partitioning the table
+///
+/// This struct also holds a full table schema to be able to cheaply hand out
+/// references to any one of the representations without needing to reconstruct them.
+#[derive(Debug, Clone)]
+pub struct TableSchema {
+    file_schema: SchemaRef,
+    table_partition_cols: Vec<FieldRef>,
+    table_schema: SchemaRef,
```

Comment:
I suggest moving this content from below.

ALso, I changed `file_schema` to `table_schema` in this comment as I think that is more accurate, but I am nore sure
```suggestion
    /// Note that the table schema is **not** the schema of the physical files.
    /// This is the schema that the physical file schema will be mapped onto,
    /// and the schema that the [`DataSourceExec`] will return.
    table_schema: SchemaRef,
```